src/sgl/device/resource.cpp

```
@@ -63,64 +63,75 @@ NativeHandle Resource::native_handle() const
-    SGL_ASSERT(m_desc.size > 0);
+    SGL_ASSERT(desc.size > 0);
 
     SGL_CHECK(
-        (m_desc.data == nullptr && m_desc.data_size == 0) || m_desc.data_size == m_desc.size,
+        (desc.data == nullptr && desc.data_size == 0) || desc.data_size == desc.size,
         "Invalid data size (got {} bytes, expected {} bytes)",
-        m_desc.data_size,
-        m_desc.size
+        desc.data_size,
+        desc.size
     );
```

Cppcheck (2.21.0)

[style] The function 'num_pages' is never used.

(unusedFunction)